models/action.go

```diff
@@ -319,6 +319,12 @@ func GetFeeds(opts GetFeedsOptions) ([]*Action, error) {
 		cond = cond.And(builder.In("repo_id", AccessibleRepoIDsQuery(opts.Actor)))
 	}
 
+	if opts.Actor == nil || !opts.Actor.IsAdmin {
+		if opts.RequestedUser.KeepActivityPrivate && actorID != opts.RequestedUser.ID {
+			return make([]*Action, 0), nil
+		}
+	}
```

Comment:
```suggestion
```

Reply:
I think we are not talking about the same expected behaviour. It is intended that admins can still see it (as shown in the UI messages). That's what causes the condition. If the user is an admin, then the whole block which could return nothing should be skipped. If the user is not an admin, then the block should be executed (second condition). However, this throws an exception if there is no actor/ the requesting user is not signed in. Due to that, it is always executed if the actor is unknown. This skips the second condition check, always runs the block and prevents the crash.

Reply:
my test setup messed up `if opts.Actor == nil || !opts.Actor.IsAdmin {` was always returning false ?!?

just considder this as resolved